backend/src/controllers/salesController.js

```diff
@@ -12,4 +15,49 @@ const getSales = (req, res) => {
   res.sendFile(filePath);
 };
 
-export { getSales };
+const getSaleById = (req, res) => {
+  const { saleId } = req.params;
+  const foundSaleRecord = sales.find((sale) => sale.saleId === saleId);
+
+  if (!foundSaleRecord) {
+    return res.status(404).json({ message: "Sale record not found" });
+  } else {
+    res.json(foundSaleRecord);
+  }
+};
+
+const createSale = (req, res) => {
+  const { productName, quantity, totalAmount, paymentMethod, status } =
+    req.body;
+
+  const requiredFields = [
+    "productName",
+    "quantity",
```

Comment:
Nice work with the validations but i think something is missing. You should not accept total amount from the req.body. What you should accept is product amount and quantity. Then the total amount is calculated here as `amount * quantity`. Imagine a store attendant can make mistake or try to steal from the store and put the wrong total amount here.